Userland/Services/WindowServer/Window.cpp

```diff
@@ -516,20 +517,31 @@ void Window::handle_keydown_event(KeyEvent const& event)
         return;
     }
     if (event.modifiers() == Mod_Alt && event.code_point() && m_menubar.has_menus()) {
-        Menu* menu_to_open = nullptr;
-        m_menubar.for_each_menu([&](Menu& menu) {
-            if (to_ascii_lowercase(menu.alt_shortcut_character()) == to_ascii_lowercase(event.code_point())) {
-                menu_to_open = &menu;
-                return IterationDecision::Break;
+        // When handling alt shortcuts, we only care about the key that has been pressed in addition
+        // to alt, not the code point that has been mapped to alt+[key], so we have to look up the
+        // scancode directly from the "unmodified" character map.
+        auto character_map_or_error = Keyboard::CharacterMap::fetch_system_map();
+        if (!character_map_or_error.is_error()) {
+            auto& character_map = character_map_or_error.value();
+
+            auto index = event.scancode() & 0xff;
```

Comment:
this clipping is about ensuring that scan code is less than 256 or just taking first byte of the scan code?

(iʼd guess that thereʼre no two byte scan codes but if thereʼre are, then it means we conflate some of the codes… is that good then? 🤔)

Reply:
The latter, that's how the kernel does it as well:

https://github.com/SerenityOS/serenity/blob/d5c913082b5efd7f8985ba9b76979edb28bc507a/Kernel/Devices/HID/HIDManagement.cpp#L181

Reply:
care to point that out in the code? (itʼd be nice to point to the kernel code but such references will get lost pretty soon!) no problem to ignore that suggestion, ofc!

Reply:
Right, I've added a comment similar to the one in the kernel.